### PR TITLE
:lipstick: Created content development hero section #156

### LINKED ISSUE #156: 🎨 Create Content Development Hero Section
#  Create the content development hero section 
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a nice hero section on the content development page that captures my attention and introduces the page content. Reuse the text-content component created in #20, and the reusable slider button in #24. I want a good experience on the web and phone.

Design:

content development header

Link to full design:

```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528/
```
 
### Technical Analysis

This feature can be found in libs/features/pages/content-development inside a components folder. The component's name is content-dev-hero-section.

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 30 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the global style.scss
5. The design is responsive for screen, tablet and phone

### PR DESCRIPTION
**Pull request by levy-web**
# Description

created a hero section for content developmet page

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #156

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Screenshot (optional)
Screenshot from 2023-09-04 05-58-49

Screenshot from 2023-09-04 06-17-41

Screenshot from 2023-09-04 05-58-18

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Beulah-Matt — CHANGES_REQUESTED** [review]
Attach your screenshots and update the PR accordingly

**Comment by levy-web** [comment]
Hi @Beulah-Matt I have updated the PR

File: libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.scss [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+}
+
+.slider-button{
+    margin-top: 57px;
+    z-index: 5999999999090999999999999999;
+}
+
+.cont-one{
+    background-color: #F2CF73;
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.scss:50]: For color/background color styling, ensure the use of CSS variables
```
background-color: var(--elewa-website-hero-section-background-color);
```

File: libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.scss [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    margin-top: auto;
+    mix-blend-mode: multiply;   
+    z-index: 2;
+    clip-path: inset(0 0 0 round 30vh);
+    background: var(--elewa-website-hero-section-cont-designs-background)
+}
+
+.cont-two{
+    background-color: #F2CF73;
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.scss:59]: Same here

File: libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.scss [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    margin-top: auto;
+    mix-blend-mode: multiply;    
+    z-index: 2;
+    clip-path: inset(0 0 0 round 30vh);
+    background: var(--elewa-website-hero-section-cont-designs-background)
+}
+
+.cont-three{
+    background-color: #F2CF73;
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.scss:68]: Same here

File: libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+export class ContentDevHeroSectionComponent {
+  heroTexts: HeroText = {
+    title:"Content development",
+    description:"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor dolorem ipsum quia dolor sit Eos qui ratione voluptatem sequi nesciunt."
+  }
+
+  buttonData: SliderButtonData = {
+    text: "unlock your potential"
+  }
+
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.ts:20]: To keep the component lean and ensure readability and maintainability, the 'heroTexts' variable can be moved to its own Typescript file in the data module.

**Comment by levy-web** [comment]
@AnnaL001 fixed the issues kindly review

File: libs/features/pages/content-development/src/lib/main/content-development-page/content-development-page.component.scss [hunk 5]
```diff
@@ -0,0 +1,2 @@
+
+
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/main/content-development-page/content-development-page.component.scss:2]: For cosmetic purposes, the whitespace can be removed

**Review by AnnaL001 — CHANGES_REQUESTED** [review]
Good job so far, just a couple of changes before this is merged

File: libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  selector: 'elewa-website-content-dev-hero-section',
+  templateUrl: './content-dev-hero-section.component.html',
+  styleUrls: ['./content-dev-hero-section.component.scss'],
+})
+export class ContentDevHeroSectionComponent {
+  heroTexts = __heroTexts
+
+  buttonData: SliderButtonData = {
+    text: "unlock your potential"
+  }
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/components/content-dev-hero-section/content-dev-hero-section.component.ts:16]: Nicely done, just one more change. You can also move the buttonData to the same file as that of the 'heroTexts' variable and reference it the same way as that of the 'heroTexts' variable.

**Comment by levy-web** [comment]
@AnnaL001 done with the changes

**Review by AnnaL001 — DISMISSED** [review]

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**